core/src/test/java/edu/wpi/grip/core/serialization/ProjectTest.java

```diff
@@ -131,6 +132,39 @@ public void testSerializeEmptyPipeline() throws Exception {
   }
 
   @Test
+  public void testLoadInvalidXML() {
+    try {
+      project.open("");
+    } catch (InvalidSaveException e) {
+      if (!e.getMessage().equals("Invalid XML")) {
+        // Not the exception we were expecting, throw it
+        throw e;
+      }
+    }
+  }
+
+  @Test
+  public void testLoadIncompatibleEntries() {
+    String xml = "<grip:Project>\n"
+        + "  <grip:Pipeline>\n"
+        + "    <sources/>\n"
+        + "    <steps>\n"
+        + "      <grip:Step name=\"\"/>\n"
+        + "    </steps>\n"
+        + "    <connections/>\n"
+        + "    <settings/>\n"
+        + "  </grip:Pipeline>"
+        + "</grip:Project>";
```

Comment:
This makes me wish for kotlin multiline strings:
https://kotlinlang.org/docs/reference/basic-types.html#string-literals